src/tiles/tiles.ts

```diff
@@ -363,6 +364,18 @@ export function geometryToTileRanges<G extends GeoJSONGeometry>(
       return [boundingBoxToTileRange(geometry, zoom, metatile, referenceTileGrid)];
     case geometry instanceof Polygon:
       return polygonToTiles(geometry, zoom, metatile, referenceTileGrid);
+    case geometry instanceof GeometryCollection: {
+      // tile ranges may overlap for overlapping geometries inside geometry collection
+      const tileRanges = geometry.geometries
+        .flatMap(flatGeometryCollection)
+        .filter((geometry): geometry is GeoJSONPolygon => geometry.type === 'Polygon' && Array.isArray(geometry.coordinates))
```

Comment:
Is there an `enum` for `geometry.type`?

Reply:
nope, but since it's union of string literals you can avoid enums